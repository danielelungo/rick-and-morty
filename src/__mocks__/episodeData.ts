export const episodeData = [
  { date: "December 2, 2013", id: 1, name: "Pilot", episode: "S01E01" },
  { date: "December 9, 2013", id: 2, name: "Lawnmower Dog", episode: "S03E03" },
];
