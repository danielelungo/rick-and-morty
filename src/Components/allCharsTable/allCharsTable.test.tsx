import React from "react";
import Char from "./allCharsTable";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { data } from "../../__mocks__/charData";
import { render, screen } from "@testing-library/react";

describe("AllCharsTable component", () => {
  test("renders AllCharsTable Component", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Char data={data} page={1} />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders correct Name, Status and Species text", () => {
    render(
      <BrowserRouter>
        <Char data={data} page={1} />
      </BrowserRouter>,
    );
    const nameText = screen.getByText(data[0].name);
    const statusText = screen.getByText(data[0].status);
    const speciesText = screen.getByText(data[0].species);
    expect(nameText).toBeInTheDocument();
    expect(statusText).toBeInTheDocument();
    expect(speciesText).toBeInTheDocument();
  });
  test("renders error if data.length is 0", () => {
    render(
      <BrowserRouter>
        <Char data={[]} page={1} />
      </BrowserRouter>,
    );
    const errorText = screen.getByText(/There is no data/);
    expect(errorText).toBeInTheDocument();
  });
});
