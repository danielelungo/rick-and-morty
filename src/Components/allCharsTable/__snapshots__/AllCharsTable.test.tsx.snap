// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllCharsTable component renders AllCharsTable Component 1`] = `
<div>
  <div
    className="sc-bczRLJ jIDRa"
    onClick={[Function]}
  >
    <div
      className="sc-gsnTZi ejnaRx"
    >
      Rick Sanchez
    </div>
    <div
      className="sc-dkzDqf iJjjTb"
      color="Alive"
    >
      Alive
    </div>
    <div
      className="sc-hKMtZM dfSjXw"
    >
      Human
    </div>
  </div>
</div>
`;
